ethcore/light/src/client/header_chain.rs

```diff
@@ -230,7 +390,7 @@ impl HeaderChain {
 					best.hash
 				};
 
-				self.headers.read().get(&hash).cloned()
+				load_from_db(hash)
```

Comment:
Happy if this is added later. The closer is the cache to the code that uses it, the better is performance. Introducing multiple levels of caching can bring huge benefits. Higher level cache usually holds less data but is accessed much more frequently. CPU hardware design generally follows this principle and it often applies to the software as well.